From 69dba60c45943050241078d7d654469c81bf41b5 Mon Sep 17 00:00:00 2001
From: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
Date: Tue, 26 May 2020 03:36:17 +0000
Subject: [PATCH] syscalls: Ignore those CMD customized for 32-bit system in
 64-bit only system

When CONFIG_64BIT is defined and CONFIG_COMPAT is not defined, the fcntl system call
"F_GETLK64", "F_SETLK64" and "F_SETLKW64" should be ignored.

Upstream-Status: Pending

Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
Signed-off-by: Yanfei Xu <yanfei.xu@windriver.com>
---
 include/instrumentation/syscalls/headers/syscalls_integers_override.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/instrumentation/syscalls/headers/syscalls_integers_override.h b/include/instrumentation/syscalls/headers/syscalls_integers_override.h
index da548b1..406ddb2 100644
--- a/include/instrumentation/syscalls/headers/syscalls_integers_override.h
+++ b/include/instrumentation/syscalls/headers/syscalls_integers_override.h
@@ -38,9 +38,13 @@ SC_LTTNG_TRACEPOINT_ENUM(lttng_fcntl_cmd_flags,
 		ctf_enum_value("F_GETOWN", F_GETOWN)
 		ctf_enum_value("F_SETSIG", F_SETSIG)
 		ctf_enum_value("F_GETSIG", F_GETSIG)
+
+#if !defined(CONFIG_64BIT) || defined(CONFIG_COMPAT)
 		ctf_enum_value("F_GETLK64", F_GETLK64)
 		ctf_enum_value("F_SETLK64", F_SETLK64)
 		ctf_enum_value("F_SETLKW64", F_SETLKW64)
+#endif
+
 		ctf_enum_value("F_SETOWN_EX", F_SETOWN_EX)
 		ctf_enum_value("F_GETOWN_EX", F_GETOWN_EX)
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,6,0))
-- 
